server/src/main/java/org/elasticsearch/index/mapper/JsonFieldMapper.java

```diff
@@ -267,6 +267,21 @@ public Query existsQuery(QueryShardContext context) {
             return new PrefixQuery(term);
         }
 
+        @Override
+        public Query rangeQuery(Object lowerTerm,
+                                Object upperTerm,
+                                boolean includeLower,
+                                boolean includeUpper,
+                                QueryShardContext context) {
+            if (lowerTerm == null || upperTerm == null) {
```

Comment:
Implemented naively, a range query with only one bound could incorrectly match values from other keys in the object. For example, a query with only a lower bound would become `(first\0value, null)`, and would match also values whose key is `second`.

We could fix this, but I wasn't sure the extra complexity was worth it. It's not clear to me how useful range queries will be right now, without support for numerics.

Reply:
I feel like this limitation is ok for now. We can revisit this later if necessary